tests/flows/native_token_bridge.go

```diff
@@ -0,0 +1,383 @@
+		valueToReturn           = big.NewInt(0).Div(valueToSend, big.NewInt(4))
+		ctx                     = context.Background()
+		deployerAddress         = common.HexToAddress("0x1337cfd2dCff6270615B90938aCB1efE79801704")
+		tokenReceiverAddress    = common.HexToAddress("0x0123456789012345678901234567890123456789")
+		burnedTxFeeAddress      = common.HexToAddress("0x0100000000000000000000000000000000000000")
+
+		emptyDestFeeInfo = nativetokendestination.TeleporterFeeInfo{
+			FeeTokenAddress: common.Address{},
+			Amount:          common.Big0,
+		}
+		emptySourceFeeInfo = nativetokensource.TeleporterFeeInfo{
+			FeeTokenAddress: common.Address{},
+			Amount:          common.Big0,
+		}
+	)
+
+	subnets := network.GetSubnetsInfo()
+	subnetA := subnets[0]
+	subnetB := subnets[1]
+	teleporterContractAddress := network.GetTeleporterContractAddress()
+
+	// Info we need to calculate for the test
+	deployerPK, err := crypto.HexToECDSA(deployerKeyStr)
+	Expect(err).Should(BeNil())
+	bridgeContractAddress, err := deploymentUtils.DeriveEVMContractAddress(deployerAddress, 0)
+	Expect(err).Should(BeNil())
+	log.Info("Native Token Bridge Contract Address: " + bridgeContractAddress.Hex())
+
+	{ // Deploy the contracts
+		// Both contracts in this test will be deployed to 0xAcB633F5B00099c7ec187eB00156c5cd9D854b5B,
+		// though they do not necessarily have to be deployed at the same address, each contract needs
+		// to know the address of the other.
+		// The nativeTokenDestination contract must be added to "adminAddresses" of "contractNativeMinterConfig"
+		// in the genesis file for the subnet. This will allow it to call the native minter precompile.
+		erc20TokenSourceAbi, err := nativetokensource.NativeTokenSourceMetaData.GetAbi()
+		Expect(err).Should(BeNil())
+		utils.DeployContract(
+			ctx,
+			NativeTokenSourceByteCodeFile,
+			deployerPK,
+			subnetA,
+			erc20TokenSourceAbi,
+			teleporterContractAddress,
+			subnetB.BlockchainID,
+			bridgeContractAddress,
+		)
+		Expect(err).Should(BeNil())
+
+		nativeTokenDestinationAbi, err := nativetokendestination.NativeTokenDestinationMetaData.GetAbi()
+		Expect(err).Should(BeNil())
+		utils.DeployContract(
+			ctx,
+			NativeTokenDestinationByteCodeFile,
+			deployerPK,
+			subnetB,
+			nativeTokenDestinationAbi,
+			teleporterContractAddress,
+			subnetA.BlockchainID,
+			bridgeContractAddress,
+			initialReserveImbalance,
+		)
+
+		log.Info("Finished deploying Bridge contracts")
+	}
+
+	// Create abi objects to call the contract with
+	nativeTokenDestination, err := nativetokendestination.NewNativeTokenDestination(
+		bridgeContractAddress,
+		subnetB.WSClient,
+	)
+	Expect(err).Should(BeNil())
+	nativeTokenSource, err := nativetokensource.NewNativeTokenSource(
+		bridgeContractAddress,
+		subnetA.WSClient,
+	)
+	Expect(err).Should(BeNil())
+
+	// Helper function
+	sendTokensToSource := func(valueToSend *big.Int, fromKey *ecdsa.PrivateKey, toAddress common.Address) *types.Receipt {
```

Comment:
Can we have these helper functions as standalone functions at the bottom of this file?

Reply:
I think I slightly prefer it the way it is. The functions use several variables local to the test, which is why they're defined inside the test. We'd have to add `ctx`, `network`, `subnetA`, `subnetB`, and the token contract to the parameters in order to extract it.